### PR TITLE
Semver tool type definition test implementation

### LINKED ISSUE #36182: Semver tool test for Type definition
List down the possible positive + negative unit test cases to validate semantic CLI tool type definition APIs.
Need to implement those unit test cases for patch version verification.

split test data sets of module type definition and keep separate test sets in a separate json file for sub levels like annotation, documentation, identifier, type descriptor and advance type definition test cases (This is for capturing more than one changes in source code).

For an example, Rename a type definition.
Old code - `type Coord record{
int x;
int y;
 };
`
New code - `type Age record{
int x;
int y;
 };
`

Here the tool should identify the change in type definition identifier and need to suggest a appropriate version change. 
Other than that tool should identify more than one change in the type definition source code.
For an example Change the name of the record and add new field in to a type definition record.
Old code - `type Person record {
	string name;
	int age;
};
`
New code - `
type Coord record{
	string name;
	int age;
       int id;
 };
`
Here tool should identify two changes and need to suggest correct version change according to that changes in the source code.

### PR DESCRIPTION
**Pull request by prathees98**
## Purpose
Semver enforcing tool ensure that the version change between two consecutive package releases based on the semantic versioning specification.
This PR is for testing the semver CLI tool for type definition.

Fixes #36182

## Approach
In this PR we have a separate json files for every typical change in type definition and compare with expected output with the current output for the test cases.
Currently the testcases of class definition covered for,

- Annotation
- Documentation
- Identifier
- Qualifier
- type descriptor
- Advance type testcases (More than one change in the source code)

## Samples
Test case format.
 {
    "description": "Remove the public keyword from a type definition",
    "oldCode": `public type Person object{
	int age;
	String name;
};,`
    "newCode": `type Person object{
	int age;
	String name;
};`
 }

## Remarks
This PR will introduce testing  semver compliance tool for Ballerina type definition.
Like that these test cases will be added in the future.

- [ ] Listener declaration
- [ ] Enumeration declaration

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/semver-checker/modules/semver-checker-core/src/test/resources/testcases/typeDefinition/advanceType.json [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                  }
+                ]
+              ]
+            }
+          ]
+        }
+      ]
+    }
+  }
+]
```
Comment (NipunaRanasinghe) [inline misc/semver-checker/modules/semver-checker-core/src/test/resources/testcases/typeDefinition/advanceType.json:112]: Missing new line

Comment (NipunaRanasinghe) [reply misc/semver-checker/modules/semver-checker-core/src/test/resources/testcases/typeDefinition/advanceType.json:112]: @prathees98 Lets update all the other similar json files as well

File: misc/semver-checker/modules/semver-checker-core/src/test/java/org/ballerinalang/semver/checker/comparator/TypeComparatorTest.java [hunk 2]
```diff
@@ -0,0 +1,527 @@
...
+    private static final String TYPE_DESCRIPTOR_TEST_ROOT = TEST_ROOT + "typeDescriptor/";
+    private static final String BEHAVIOURAL_TYPE_DESCRIPTOR_TEST_ROOT = TYPE_DESCRIPTOR_TEST_ROOT + "behaviouralTypeDescriptor/";
+    private static final String SEQUENCE_TYPE_DESCRIPTOR_TEST_ROOT = TYPE_DESCRIPTOR_TEST_ROOT + "sequenceTypeDescriptor/";
+    private static final String SIMPLE_TYPE_DESCRIPTOR_TEST_ROOT = TYPE_DESCRIPTOR_TEST_ROOT + "simpleTypeDescriptor/";
+    private static final String STRUCTURAL_TYPE_DESCRIPTOR_TEST_ROOT = TYPE_DESCRIPTOR_TEST_ROOT + "structuredTypeDescriptor/";
+    private static final String OTHER_TYPE_DESCRIPTOR_TEST_ROOT = TYPE_DESCRIPTOR_TEST_ROOT + "otherTypeDescriptor/";
+    private static final String BEHAVIOURAL_FUNCTION_TYPE_DESCRIPTOR_TEST_ROOT = BEHAVIOURAL_TYPE_DESCRIPTOR_TEST_ROOT + "function/";
+    private static final String BEHAVIOURAL_OBJECT_TYPE_DESCRIPTOR_TEST_ROOT = BEHAVIOURAL_TYPE_DESCRIPTOR_TEST_ROOT + "object/";
+    private static final String BEHAVIOURAL_OBJECT_METHOD_DEFINITION_TYPE_DESCRIPTOR_TEST_ROOT = BEHAVIOURAL_OBJECT_TYPE_DESCRIPTOR_TEST_ROOT + "methodDefinition/";
```
Comment (NipunaRanasinghe) [inline misc/semver-checker/modules/semver-checker-core/src/test/java/org/ballerinalang/semver/checker/comparator/TypeComparatorTest.java:51]: Shall we fix potential checkstyle violations in here? (line lengths should be less than 120 characters)

Comment (prathees98) [reply misc/semver-checker/modules/semver-checker-core/src/test/java/org/ballerinalang/semver/checker/comparator/TypeComparatorTest.java:51]: +1

**Review by NipunaRanasinghe — APPROVED** [review]

**Pull request merged**